### DISCUSSION TITLE
How do I implement a CategoryExtensionProvider?

### DISCUSSION THREAD

**Discussion by Snownee** [shedaniel/RoughlyEnoughItems#1282] [Q & A] [answered]
In `CategoryExtensionProvider#provide`, how do I get all the widgets from `lastView` if I don't know what the recipe bounds are?

**Comment by shedaniel** [comment, marked as answer]
Read the javadocs on the method, the extension and the view. You are supposed to make a new categoryview and delegate all methods to the lastView, then within that you may modify it, it should be similar to any visitor based system like ASM
